apps/marketplace/actions/opportunitiesActions.js

```diff
@@ -0,0 +1,56 @@
+import { OPPORTUNITIES_SUCCESS, OPPORTUNITIES_SENDING } from '../constants/constants'
+import { GENERAL_ERROR } from '../constants/messageConstants'
+import dmapi from '../services/apiClient'
+import { setErrorMessage } from './appActions'
+
+export const handleErrorFailure = response => dispatch => {
+  if (!response) {
+    dispatch(setErrorMessage(GENERAL_ERROR))
+  } else {
+    if (response.data && response.data.message) {
+      dispatch(setErrorMessage(response.data.message))
+    }
+    if (response.message) {
+      if (response.code) {
+        dispatch(setErrorMessage(`${GENERAL_ERROR} (${response.code}: ${response.message})`))
+      } else {
+        dispatch(setErrorMessage(`${GENERAL_ERROR} (${response.message})`))
+      }
+    } else {
+      setErrorMessage(GENERAL_ERROR)
+    }
+  }
+}
+
+export const sendingRequest = currentlySending => ({
```

Comment:
use `appActions.sendingRequest` ?

Reply:
I've found that the app wide `sendingRequest` interferes with this, specifically I think the "ping" request is interfering - i.e. when the ping is finished the loader indicator goes away, but opportunities data is a bit slower and isn't back in time so you get a blank body for a brief period.

A `sendingRequest` specific to opportunities means the timing is on the actual opportunities request.

Reply:
I think the real solution to this is to do the reverse - `ping` needs its own action.  Then others would be free to use `sendingRequest`

Reply:
That's a good point, although in some pages it may make sense not to separate the ping request out (if that page requires data coming from the ping response, e.g. CSRF token, user details).

Perhaps instead the app `sendingRequest` should be an array of inflight requests, that are then removed on completion?

Reply:
That could get tricky to manage - may need to think about this more